src/users/helpers.ts

```diff
@@ -0,0 +1,63 @@
+import {Request, Response, NextFunction} from "express";
+import jwt from "jsonwebtoken";
+import {getConfig} from '../config';
+
+interface IUser {
+  isAuthenticated: boolean,
+  user: { 
+    role: string, 
+    email: string 
+  } | null
+}
+
+const { jwt: {secretKey}} = getConfig();
+
+export const verifyToken = (req: Request): IUser => {
+  const { authorization = '' } = req.headers
+  const token = authorization.slice(7)
+
+  if (!token) {
+    return {
+      isAuthenticated: false,
+      user: null
+    }
+  }
+
+  try {
+    const currentUser = jwt.verify(token, secretKey) as { role: string, email: string }
+    return {
+      isAuthenticated: true,
+      user: currentUser
+    }
+  } catch (err) {
+    return {
+      isAuthenticated: false,
+      user: null
+    }
+  }
+}
+
+export const checkUserIsAdmin = (req:Request, res:Response, next: NextFunction) => {
+  const currentUser = module.exports.verifyToken(req)
+
+  if(!currentUser){
+    return res.status(403).json({message: 'Unauthorized Access! Invaild token!'})
+  } 
+  if(currentUser.isAuthenticated && (currentUser.user?.role) === 'admin'){
+    req.currentUser = currentUser.user
+    next()
+  } else {
+    res.status(403).json({message: 'Unauthorized Access! Admin only!'})
+  }
+}
+
+export const checkUserAuthenticated = (req: Request, res:Response, next: NextFunction) => {
```

Comment:
Are we checking for authentication or verification?

Reply:
Authentication after verification